compiler/util/src/org/jetbrains/kotlin/config/LanguageVersionSettings.kt

```diff
@@ -171,6 +171,8 @@ enum class LanguageFeature(
     // Looks like we can't enable it until KT-26245 is fixed because otherwise some use cases become broken because of overrides
     ProhibitUsingNullableTypeParameterAgainstNotNullAnnotated(sinceVersion = null, kind = BUG_FIX),
 
+    ExtendedTypeCompatibilityChecker(sinceVersion = null, kind = BUG_FIX, defaultState = State.DISABLED),
```

Comment:
Besides enablements in some test variants, I can't see where it is used? Also, it'd be better to have some code comments to explain what it is?

Reply:
The flag is checked at https://github.com/JetBrains/kotlin/blob/4e9f4476e0eba5228aa8b1f70d1f37175b2b1002/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/types/ConeTypeCompatibilityChecker.kt#L55. I am leaving it as disabled for now so we get FE1.0 compatible behavior. I didn't put a version there since I don't know what JetBrains folks would think. I will defer it to them.

Reply:
I missed it as well too initially because GitHub collapsed it by default, and I was wondering, where is this new flag used 😛

Reply:
I have been using IntelliJ for code reviews and it has been working pretty well. You can find more from https://blog.jetbrains.com/idea/2020/05/intellij-idea-2020-2-early-access-program-is-starting/#version_control